### PR TITLE
Add catalog CRs

### PR DESCRIPTION
**Pull request by tomahawk28**
Toward https://github.com/giantswarm/roadmap/issues/219

### CODE REVIEW

File: helm/appcatalog/templates/appcatalog.yaml [hunk 1]
```diff
@@ -2,6 +2,8 @@ apiVersion: "application.giantswarm.io/v1alpha1"
 kind: AppCatalog
 metadata:
   name: "{{ .Values.appCatalog.name }}"
+  annotations:
```
Comment (tomahawk28) [inline helm/appcatalog/templates/appcatalog.yaml:5]: When we delete `appCatalog` CRs from the helm charts later, we should keep these on clusters.

File: helm/appcatalog/templates/catalog.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+apiVersion: "application.giantswarm.io/v1alpha1"
+kind: Catalog
+metadata:
+  name: "{{ .Values.appCatalog.name }}"
+  {{- if eq .Values.appCatalog.catalogVisibility "internal" }}
```
Comment (tomahawk28) [inline helm/appcatalog/templates/catalog.yaml:5]: As the spec stated, we put all internal catalog CRs to `giantswarm` namespace. All other catalogs CRs would be located in the `default` namespace.
https://intranet.giantswarm.io/docs/architecture-specs-adrs/specs/app-catalog-scope/#namespace-structure

File: helm/appcatalog/templates/catalog.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  namespace: "giantswarm"
+  {{- else }}
+  namespace: "default"
+  {{- end }}
+  labels:
+    application.giantswarm.io/catalog-type: "{{ .Values.appCatalog.catalogType }}"
+    application.giantswarm.io/catalog-visibility: "{{ .Values.appCatalog.catalogVisibility }}"
+    app-operator.giantswarm.io/version: "{{ .Values.appOperator.version }}"
```
Comment (rossf7) [inline helm/appcatalog/templates/catalog.yaml:13]: Can we drop the version label?

The current appcatalog controller doesn't have a label selector which is OK as it's only active for the unique instance.

 Alternatively we could default it. I just don't want users to have to set it when creating their own catalogs.

**Review by rossf7 — COMMENTED** [review]
Looks good. Just a query on the version label.

File: helm/appcatalog/templates/catalog.yaml (hunk 3 above)
Comment (tomahawk28) [reply helm/appcatalog/templates/catalog.yaml:13]: Sure, good point. let's drop it.

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**